src/pages/index.js

```diff
@@ -1,107 +1,244 @@
+    // Change the button text to "Saving..."
+    const saveButton = document.querySelector("#modal-submit-button");
+    saveButton.textContent = "Saving...";
+
+    console.log("Sending card data:", cardData); // Log the card data being sent
+
+    api
+      .createCard(cardData)
+      .then((newCard) => {
+        console.log("Card added successfully:", newCard); // Log successful response
+        const cardElement = createCard(newCard, api); // Pass api to Card
+        cardSection.addItem(cardElement);
+        addCardPopup.close();
+        addCardForm.reset();
+        addCardFormValidator.disableButton();
+      })
+      .catch((error) => {
+        console.error("Error adding new card:", error); // Log any errors
+      })
+      .finally(() => {
+        saveButton.textContent = "Create"; // Reset the button text after the action
+      });
   });
   addCardPopup.setEventListeners();
 
-  /* ---------------------------- Form Validation ----------------------------- */
-  // Initialize form validators for both forms
+  const deleteConfirmationPopup = new PopupWithConfirmation(
+    "#delete-confirmation-modal",
+    (cardId) => {
+      api
+        .deleteCard(cardId)
+        .then(() => {
+          const cardElement = document.querySelector(`[data-id='${cardId}']`);
+          if (cardElement) {
+            cardElement.remove();
+          }
+          console.log("Card deleted successfully.");
+        })
+        .catch((error) => {
+          console.error("Error deleting card:", error);
+        });
+    }
+  );
+
   const profileFormValidator = new FormValidator(settings, profileEditForm);
   const addCardFormValidator = new FormValidator(settings, addCardForm);
-  // Enable form validation
   profileFormValidator.enableValidation();
   addCardFormValidator.enableValidation();
 
-  /* ---------------------------- Handlers ----------------------------- */
-  // Function to create a new Card instance
-  function createCard(cardData) {
-    const card = new Card(cardData, "#card-template", handleImageClick);
-    return card.getView();
+  // Select avatar form and initialize its validator
+  const avatarForm = document.querySelector("#profile-avatar-form");
+  const avatarFormValidator = new FormValidator(settings, avatarForm);
+  avatarFormValidator.enableValidation(); // Now correctly initialized
+
+  // Function to create a card
+  function createCard(cardData, api) {
+    const card = new Card(
+      cardData,
+      "#card-template",
+      handleImageClick,
+      api,
+      deleteConfirmationPopup
+    ); // Pass api to Card
+    const cardElement = card.getView(); // Get the card view correctly
+    cardElement.dataset.id = cardData._id;
+    // Attach event listener for the delete button after the card has been created
+    cardElement
+      .querySelector(".card__delete-button")
+      .addEventListener("click", () => {
+        deleteConfirmationPopup.open(cardData._id);
```

Comment:
Please, don't search the trash icon in `index.js`. It should be done inside `Card`

Now only class `Card` knows every **card selector**, so you need to encapsulate the likes logics inside its methods and call them in `index.js`. It’s a violation of **Single responsibility** principle

You should pass `deleteCard` function for that


```js

  function createCard(cardData, api) {
    const card = new Card(
      cardData,
      "#card-template",
      handleImageClick,
      api,
      () => {
        deleteConfirmationPopup.open(cardData._id);
      }
    ); // Pass api to Card
```